openpype/modules/kitsu/plugins/publish/integrate_kitsu_note.py

```diff
@@ -42,6 +76,12 @@ def process(self, context):
                         "changed!".format(self.note_status_shortname)
                     )
 
+            # If custom comment, create it
+            if self.custom_comment_template["enabled"]:
+                publish_comment = self.format_publish_comment(instance)
+
+            self.log.debug("Comment is `{}`".format(publish_comment))
```

Comment:
I don't think they'd want to skip if there's no comment because I think this is also the thing that actually adds the reviewable image/video as an attachment to Kitsu. (I think it is, not sure)

Reply:
Oh, really?

Reply:
It is changin status, I see....

Reply:
Oh it can be? I thought it was one comment per publish and not per instance? How would you place different comments per instance in the new publisher?

And if there is no comment we should still create a comment as it's what the review image/video later will get attached to :) As the script checks the instance for `review` in families only instance that should be published with a review will get a comment

Reply:
There is an [Instance collector](https://github.com/ynput/OpenPype/blob/develop/openpype/plugins/publish/collect_comment.py#L31) which gives option to fill comment per instance. If user won't fill it, the context comment is used. To avoid possible mismatch during publishing the collector will also fill final value to instance data to avoid these `instance.data.get("comment", instance.context.data.get("comment"))`.

The collector must be enabled and families must be defined.

NOTE: I just found out that the collector inherit from `ContextPlugin` instead of `InstancePlugin`, but it works (pyblish magic).

Reply:
@iLLiCiTiT I thought that wasn't the case for [`InstancePlugin` and `ContextPlugin` because they inherit from the `ExplicitMetaClass`](https://github.com/pyblish/pyblish-base/blob/master/pyblish/plugin.py#L320-L356). But following the logic it does seem it might still pass as long as it has either `context` or `instance` as an argument. This might actually be a bug in Pyblish since those two plug-ins were intended to be explicit.